packages/plugin-dynamic-sitemap/src/index.js

```diff
@@ -0,0 +1,30 @@
+import fs from 'fs/promises';
+
+const greenwoodPluginDynamicExport = (options = {}) => [{
```

Comment:
Interesting approach and maybe it's just something I'm missing here but the main concern / limitation I see with this approach is that there is no way to programmatically build your _own_ sitemap.  To me, the advantage of the dynamic sitemap option was that could automatically build it up whatever way you need, even tapping into your own frontmatter / page metadata.

Referencing my back of the napkin example in #1232, which is basically just taking a page out of the [Next.js playbook](https://nextjs.org/docs/app/api-reference/file-conventions/metadata/sitemap), I would expect users to be able to make a _sitemap.xml.(js|ts)_ file in the root of their workspace (e.g. _src/sitemap.xml.js_, which would just export a function in which we pass the compilation to them, and they can build it up themselves and set their own values for all [supported properties](https://www.sitemaps.org/protocol.html), since there are things they might want
```js
export async function sitemap(compilation) {
  const urls = compilation.graph.map((page) => {
    const { data, route } = route;

    return `
      <url>
        <loc>http://www.example.com${page.route}</loc>
        <changefreq>${data.freq}</changefreq>
        <priority>https://www.sitemaps.org/protocol.html#prioritydef>${data.priority ?? '1.0'}</priority>
      </url>
    `;
  });
  const sitemap = `
    <?xml version="1.0" encoding="UTF-8"?>
    <urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
      ${urls.join('\n')}
    </urlset>
  `;

  return new Response(body, { headers: 'Content-Type' : 'text/xml' })
}
```

----

As far as how to do this through greenwood, I think [the API Resource](https://github.com/ProjectEvergreen/greenwood/blob/master/packages/cli/src/plugins/resource/plugin-api-routes.js#L15) as a reference is probably going to get us halfway there at least for development, such that hitting `https://localhost:1984/sitemap.xml` returns something.

For build time, we maybe add something to the build step [like we do for adapters](https://github.com/ProjectEvergreen/greenwood/blob/master/packages/cli/src/commands/build.js#L120), where if we have a sitemap (perhaps the graph can track this on init?) we `import` the that function from the user's file and write out the file for them?